packages/amplify_core/lib/src/types/api/types/rest/rest_operation.dart

```diff
@@ -23,7 +23,7 @@ class RestOperation extends AWSHttpOperation<AWSHttpResponse> {
           return response;
         }
         // In case other response types ever added.
-        throw const ApiException('Unable to convert to AWSHttpResponse');
+        throw const ApiOperationError('Unable to convert to AWSHttpResponse');
```

Comment:
Better to throw a `StateError` here. Exceptions are used for things which are expected. If we've forgotten to update this code - we messed up and this should fail hard and never be released to prod.